### ISSUE TITLE
Add Minecraft Pi

### ISSUE THREAD

**Issue by Botspot** [Botspot/pi-apps#144]
This is a continuation of #137.

Any new conversation about adding MC Pi to Pi-Apps should be done here.

**Comment by zachdwilliams** [comment]
Will there be Hardcore mode?

**Comment by TheBrokenRail** [comment]
> Will there be Hardcore mode?

No

**Comment by zachdwilliams** [comment]
Alright then

**Comment by zachdwilliams** [comment]
Can you call it “Minecraft Pi Launcher” so it won’t be confused with the pre-installed Minecraft Pi on Raspbian?

**Comment by Botspot** [comment]
Error output:
```
2020-12-26 21:55:57 (4.63 MB/s) - ‘/home/pi/pi-apps/apps/Minecraft Pi/debs.zip’ saved [222841877]

unzip:  cannot find or open /home/pi/pi-apps/apps/Minecraft Pi/deb.zip, /home/pi/pi-apps/apps/Minecraft Pi/deb.zip.zip or /home/pi/pi-apps/apps/Minecraft Pi/deb.zip.ZIP.
Unable To Unzip

Failed to install Minecraft Pi with install script!

Close this window to exit.
```

**Comment by Botspot** [comment]
Also @TheBrokenRail, it appears your script only installs docker if the OS is `buster`. Why? If the OS was something other than Buster, wouldn't that mean that docker would not be installed?

Another issue. After installing once, then uninstalling, then reinstalling again, this error appears:
```
# Executing docker install script, commit: 3d8fe77c2c46c5b7571f94b42793905e5b3e42e4
Warning: the "docker" command appears to already exist on this system.

If you already have Docker installed, this script can cause trouble, which is
why we're displaying this warning and provide the opportunity to cancel the
installation.

If you installed the current Docker package using this script and are using it
again to update Docker, you can safely ignore this message.

You may press Ctrl+C now to abort this script.
+ sleep 20
```

**Comment by Botspot** [comment]
Now I see this:
```
mkdir: cannot create directory ‘/home/pi/pi-apps/apps/Minecraft Pi/deb’: File exists

Failed to install Minecraft Pi with install script!
```

**Comment by Botspot** [comment]
After fixing that bug, now I see:
```
dpkg: error: cannot access archive '/home/pi/pi-apps/apps/Minecraft Pi/deb/deb
/home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb
/home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-virgl_1.0.0.20201225.1901_all.deb
/home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-server_1.0.0.20201225.1902_all.deb': No such file or directory
Unable To Install

Failed to install Minecraft Pi with install script!
```

**Comment by Botspot** [comment]
After making substantial changes, I seem stuck here:
```
Installing Minecraft Pi with install script
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package 'docker.io' is not installed, so not removed
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 minecraft-pi-server : Depends: libseccomp2 (>= 2.4.2) but 2.3.3-4 is to be installed
                       Depends: docker-compose but it is not going to be installed
                       Depends: qemu-user-static but it is not going to be installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
/home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb
/home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-virgl_1.0.0.20201225.1901_all.deb
/home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-server_1.0.0.20201225.1902_all.deb
Selecting previously unselected package minecraft-pi-native.
dpkg: regarding .../minecraft-pi-native_1.0.0.20201225.1901_all.deb containing minecraft-pi-native:
 minecraft-pi-native conflicts with minecraft-pi
  minecraft-pi (version 0.1.1-6) is present and installed.

dpkg: error processing archive /home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb (--install):
 conflicting packages - not installing minecraft-pi-native
Selecting previously unselected package minecraft-pi-virgl.
dpkg: regarding .../minecraft-pi-virgl_1.0.0.20201225.1901_all.deb containing minecraft-pi-virgl:
 minecraft-pi-virgl conflicts with minecraft-pi
  minecraft-pi (version 0.1.1-6) is present and installed.

dpkg: error processing archive /home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-virgl_1.0.0.20201225.1901_all.deb (--install):
 conflicting packages - not installing minecraft-pi-virgl
(Reading database ... 273061 files and directories currently installed.)
Preparing to unpack .../minecraft-pi-server_1.0.0.20201225.1902_all.deb ...
Unpacking minecraft-pi-server (1.0.0.20201225.1902) over (1.0.0.20201225.1902) ...
dpkg: dependency problems prevent configuration of minecraft-pi-server:
 minecraft-pi-server depends on libseccomp2 (>= 2.4.2); however:
  Version of libseccomp2:armhf on system is 2.3.3-4.
 minecraft-pi-server depends on docker-compose; however:
  Package docker-compose is not installed.
 minecraft-pi-server depends on qemu-user-static; however:
  Package qemu-user-static is not installed.

dpkg: error processing package minecraft-pi-server (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 /home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb
 /home/pi/pi-apps/apps/Minecraft Pi/deb/deb/minecraft-pi-virgl_1.0.0.20201225.1901_all.deb
 minecraft-pi-server
Unable To Install

Failed to install Minecraft Pi with install script!

Close this window to exit.
```

**Comment by Botspot** [comment]
Here's the current app I've made changes to:  
[Minecraft Pi.zip](https://github.com/Botspot/pi-apps/files/5744084/Minecraft.Pi.zip)

**Comment by Botspot** [comment]
Added with commit 4d2bb0bc0a5e365d85bbdb766f58806dd0df8a4d.

**Comment by TheBrokenRail** [comment]
The script you added installs the abandoned MCPIL, this should probably be replaced with MCPIL-R. The uninstall script also hasn't been updated. MCPIL-R requires MCPI-Docker (the thing my script installed) and is available [here](https://github.com/MCPI-Devs/MCPIL-R).

**Comment by TheBrokenRail** [comment]
How is this?

[Minecraft Pi.zip](https://github.com/Botspot/pi-apps/files/5815517/Minecraft.Pi.zip)

**Comment by mobilegmYT** [comment]
Ah, probably just close this pull request and use tbr's one mentioned above @Botspot.

**Comment by Pythonic456** [comment]
@Botspot What survival mode? It says in the description that there should be survival mode, but how do I use it?
Thanks :)

**Comment by TheBrokenRail** [comment]
> @Botspot What survival mode? It says in the description that there should be survival mode, but how do I use it?
> Thanks :)

When using MCPI-Docker/MCPIL-R, you should just have an option when creating a new world.

**Comment by Botspot** [comment]
@TheBrokenRail Your script (that I just pushed) seems to be having problems:
```
Installing Minecraft Pi with install script
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Package 'docker.io' is not installed, so not removed
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Executing: /tmp/apt-key-gpghome.XCnwGXv4Dc/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 04EE7237B7D453EC
gpg: key E0B11894F66AEC98: "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
Executing: /tmp/apt-key-gpghome.F4V3qlNIy1/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys 648ACFD622F3D138
gpg: key DC30D7C23CBBABEE: "Debian Archive Automatic Signing Key (10/buster) <ftpmaster@debian.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
deb http://deb.debian.org/debian buster-backports main
Hit:1 http://archive.raspberrypi.org/debian buster InRelease
Hit:2 https://download.docker.com/linux/raspbian buster InRelease              
Get:3 http://raspbian.raspberrypi.org/raspbian buster InRelease [15.0 kB]      
Hit:4 https://download.mono-project.com/repo/debian stable-raspbianbuster InRelease
Hit:5 http://plugwash.raspbian.org/sonic-pi buster-sonicpi InRelease           
Ign:6 http://repo.vivaldi.com/stable/deb stable InRelease                      
Hit:7 http://repo.vivaldi.com/stable/deb stable Release                        
Ign:9 https://download.puffinbrowser.com/repo stable InRelease                 
Hit:10 https://download.puffinbrowser.com/repo stable Release                  
Hit:8 https://packagecloud.io/Alvarito050506/mcpi-devs/raspbian buster InRelease
Hit:12 http://deb.debian.org/debian buster-backports InRelease                 
Get:14 https://repo.steampowered.com/steam stable InRelease [2,852 B]
Get:15 https://repo.steampowered.com/steam stable/steam Sources [564 B]        
Get:16 https://repo.steampowered.com/steam stable/steam i386 Packages [868 B]  
Get:17 https://repo.steampowered.com/steam stable/steam amd64 Packages [672 B] 
Fetched 19.9 kB in 8s (2,603 B/s)                                              
Reading package lists... Done
W: Target Packages (main/binary-armhf/Packages) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Translations (main/i18n/Translation-en_US) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Translations (main/i18n/Translation-en_US.UTF-8) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11 (main/dep11/Components-armhf.yml) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Packages (main/binary-armhf/Packages) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Translations (main/i18n/Translation-en_US) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target Translations (main/i18n/Translation-en_US.UTF-8) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11 (main/dep11/Components-armhf.yml) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:42 and /etc/apt/sources.list:43
Warning: wildcards not supported in HTTP.
--2021-01-14 11:56:40--  https://jenkins.thebrokenrail.com/job/minecraft-pi-docker/job/master/lastSuccessfulBuild/artifact/out/deb/*zip*/deb.zip
Resolving jenkins.thebrokenrail.com (jenkins.thebrokenrail.com)... 2601:40a:8280:ce20::1, 68.43.64.50
Connecting to jenkins.thebrokenrail.com (jenkins.thebrokenrail.com)|2601:40a:8280:ce20::1|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/zip]
Saving to: ‘/home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb.zip’

/home/pi/pi-apps/ap     [              <=>   ] 212.52M  2.93MB/s    in 67s     

2021-01-14 11:57:47 (3.19 MB/s) - ‘/home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb.zip’ saved [222841877]

Archive:  /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb.zip
  inflating: /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb  
  inflating: /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb/deb/minecraft-pi-server_1.0.0.20201225.1902_all.deb  
  inflating: /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb/deb/minecraft-pi-virgl_1.0.0.20201225.1901_all.deb  
Installing /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb...
dpkg: regarding .../minecraft-pi-native_1.0.0.20201225.1901_all.deb containing minecraft-pi-native:
 minecraft-pi-native conflicts with minecraft-pi
  minecraft-pi (version 0.1.1-6) is present and installed.

dpkg: error processing archive /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb (--install):
 conflicting packages - not installing minecraft-pi-native
Errors were encountered while processing:
 /home/pi/pi-apps/apps/Minecraft Pi/mcpi-docker/deb/deb/minecraft-pi-native_1.0.0.20201225.1901_all.deb

Failed to install Minecraft Pi with install script!

Close this window to exit.
```

**Comment by Botspot** [comment]
Here's an updated version of the Minecraft Pi app that fixes a bunch of bugs:  
[Minecraft Pi.zip](https://github.com/Botspot/pi-apps/files/5817732/Minecraft.Pi.zip)

**Comment by TheBrokenRail** [comment]
Here is an updated version tat replaces ``apt`` with ``apt-get``:  [Minecraft Pi.zip](https://github.com/Botspot/pi-apps/files/5822100/Minecraft.Pi.zip)

**Comment by Pythonic456** [comment]
@Botspot I'm happy to test

**Comment by mobilegmYT** [comment]
Here's also updated icons from the latest version.
![icon-24](https://user-images.githubusercontent.com/44128563/104859452-b0620880-58e2-11eb-875b-775fbbc245ec.png)
![icon-64](https://user-images.githubusercontent.com/44128563/104859453-b0fa9f00-58e2-11eb-9ff6-220ec5c1a835.png)

**Comment by Botspot** [comment]
@TheBrokenRail I just realized something:
```
# Setup Backports Repo On Buster
if grep -q '^deb http://deb.debian.org/debian buster-backports main$' /etc/apt/sources.list; then
```
That right there does the exact opposite of what you want. If grep sees no match, the if statement will not run. So it won't set up the backports repository. On the other hand, if grep does detect a match already there, then it will setup the backports repo **again**.

Simple fix: add a `!`:
```
if ! grep -q '^deb http://deb.debian.org/debian buster-backports main$' /etc/apt/sources.list; then
```

**Comment by Botspot** [comment]
Officially released on Pi-Apps with commit 3accefd316e473791fd1b0b786d4e88e4a347bbf.
